packages/flutter/lib/src/services/asset_manifest.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'asset_bundle.dart';
+import 'message_codecs.dart';
+
+const String _kAssetManifestFilename = 'AssetManifest.bin';
+
+/// Contains details about available assets.
+abstract class AssetManifest {
+  /// Loads asset manifest data from an [AssetBundle] object and creates an
+  /// [AssetManifest] object from that data.
+  static Future<AssetManifest> loadFromAssetBundle(AssetBundle bundle) {
+    return bundle.loadStructuredBinaryData(_kAssetManifestFilename, _AssetManifestBin.fromStandardMessageCodecMessage);
+  }
+
+  /// Loads asset manifest data from the root bundle and creates an
+  /// [AssetManifest] from that data.
+  static Future<AssetManifest> loadFromRootBundle() {
+    return loadFromAssetBundle(rootBundle);
+  }
+
+  /// Lists the keys of all known assets, not including asset variants.
+  Iterable<String> listAssets() {
+    throw UnimplementedError();
+  }
+
+  /// Gets available variants of an asset.
+  Iterable<AssetVariant> getAssetVariants(String key) {
```

Comment:
I guess someone could modify it and it would mess with the cache though 🤷‍♂️

Reply:
> I guess someone could modify it and it would mess with the cache though 🤷‍♂️

Yeah. I think we don't have readonly lists out-of-the-box in Dart. I suppose we could return a new copy of the list for every call--I imagine that, for the majority of use-cases, it wouldn't be terribly expensive in speed or memory. I still prefer returning `Iterable` whenever I have any doubts, but I don't know if this is a norm in the Flutter API.

Reply:
No, generally speaking we return a List. if its small copying is a fine choice